std/net/curl.d

```diff
@@ -2330,9 +2330,9 @@ private bool decodeLineInto(Terminator, Char = char)(ref const(ubyte)[] basesrc,
   * http.perform();
   *
   * // Track progress
+  * auto http = HTTP();
```

Comment:
The parts of this example above do not redeclare `http`.

Reply:
I though it did redeclared. 
I'm sorry then.

Reply:
OK, then either this should be removed again, or this example split up into several examples using `---` delimiters for each block.

Reply:
I think it would be better to spilt up into two examples. We shall do that.

Reply:
- You need to add a terminating `---` in addition to the starting `---`.
- You should move the comments from the start of each example to a DDoc paragraph.
- The import statements at the top of the first example should be duplicated in every example.
- There should probably be two examples, not three, one per each request.

Reply:
Okay, I spilt it into three examples. I'm keep on hoping I didn't make more mistakes/errors.

Reply:
Whoops, still failing.

Reply:
Please test that the separate examples work. For example I see the second one is missing a `http` declaration.

Reply:
All examples work except this one:

I tested this one Manually, and it seems that there might be problems with **curl**, it is throwing exceptions.

https://github.com/dlang/phobos/blob/e775bd9973c8135c0e2881ec914d0d98fc9a4838/std/net/curl.d#L2320-L2341

```
std.net.curl.CurlException@std\net\curl.d(4364): URL using bad/illegal format or missing URL on handle 842560
----------------
0x004159E1
0x00402362
0x00403E27
0x00403DA1
0x00403C3C
0x004024DF
0x770B6359 in BaseThreadInitThunk
0x77977A94 in RtlGetAppContainerNamedObjectPath
0x77977A64 in RtlGetAppContainerNamedObjectPath
```

Reply:
> missing URL on handle

You did not specify an URL for the request.

Note how the first example specifies an URL as the parameter to the `HTTP` constructor.

Reply:
Yep, I noticed that. But it still throws error. The host is probably responding, otherwise it would throw exception "`Couldn't resolve host name on handle`".

```
std.net.curl.CurlException@std\net\curl.d(4364): Send failed since rewinding of the data stream failed on handle 9A2560
----------------
0x00414DD9
0x0040236A
0x00403E2F
0x00403DA9
0x00403C44
0x004024E7
0x770B6359 in BaseThreadInitThunk
0x77977A94 in RtlGetAppContainerNamedObjectPath
0x77977A64 in RtlGetAppContainerNamedObjectPath
```

Reply:
I don't know why you get that error. Perhaps you made a typo? Can you post the entire program you're trying to run?

Reply:
```
import std.net.curl, std.stdio;

void main(){
 auto http = HTTP("dlang.org");
 auto msg = "Hello world";
 http.contentLength = msg.length;
 http.onSend = (void[] data)
 {
     auto m = cast(void[]) msg;
     size_t len = m.length > data.length ? data.length : m.length;
     if (len == 0) return len;
     data[0 .. len] = m[0 .. len];
     msg = msg[len..$];
     return len;
 };
 http.perform();
}
```

Reply:
Strange that you get "couldn't resolve host name". Perhaps it is a peculiarity of curl and SSL requests on WIndows. I get a different error message (`Send failed since rewinding of the data stream failed on handle ...`), as mentioned above fixed by changing the URL to use HTTPS.

Reply:
That was my mistake, I should get proper text editor. 
```
std.net.curl.CurlException@std\net\curl.d(4364): Send failed since rewinding of the data stream failed on handle C32560
```

Reply:
OK. Works perfectly.